### PR TITLE
CMake Refactor in Xtp

### PR DESCRIPTION
**Pull request by junghans**
Part of https://github.com/votca/votca/pull/160

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,9 +33,6 @@ include(CheckCXXCompilerFlag)
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF) #...without compiler extensions like gnu++11
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel" AND CMAKE_VERSION VERSION_LESS "3.6.0")
```
Comment (JensWehner) [inline CMakeLists.txt]: why did you remove this?

Comment (junghans) [reply CMakeLists.txt]: Because we bumped the minimum CMake version to 3.10.
